examples/matrix_multiplier/tests/test_matrix_multiplier.py

```diff
@@ -0,0 +1,181 @@
+import os
+from random import getrandbits
+
+import cocotb
+from cocotb.binary import BinaryValue
+from cocotb.clock import Clock
+from cocotb.monitors import BusMonitor
+from cocotb.regression import TestFactory
+from cocotb.scoreboard import Scoreboard
+from cocotb.triggers import RisingEdge, ReadOnly
+
+NUM_SAMPLES = int(os.environ.get('NUM_SAMPLES', 3000))
+DATA_WIDTH = int(os.environ['DATA_WIDTH'])
+A_ROWS = int(os.environ['A_ROWS'])
+B_COLUMNS = int(os.environ['B_COLUMNS'])
+A_COLUMNS_B_ROWS = int(os.environ['A_COLUMNS_B_ROWS'])
+
+
+class MatMultMonitor(BusMonitor):
+
+    _signals = ["A", "B", "i_valid", "C", "o_valid"]
+
+    def __init__(self, dut, multi_dimension, callback=None, event=None):
+        self.multi_dimension = multi_dimension
+        self.clock = dut.clk
+        BusMonitor.__init__(self, dut, None, dut.clk, callback=callback, event=event)
+
+    def _unpack_matrix(self, signal, rows, cols):
+        if self.multi_dimension:
+            return signal.value
+        else:
+            return [[signal[(i * cols) + j].value for j in range(cols)] for i in range(rows)]
+
+    @cocotb.coroutine
+    def _monitor_recv(self):
+
+        in_values = None
+
+        while True:
+            yield RisingEdge(self.clock)
+            yield ReadOnly()
+
+            if self.bus.o_valid.value and in_values:
+                self._recv((in_values[0], in_values[1], self._unpack_matrix(self.bus.C, A_ROWS, B_COLUMNS)))
+                in_values = None
+
+            if self.bus.i_valid.value:
+                in_values = (
+                    self._unpack_matrix(self.bus.A, A_ROWS, A_COLUMNS_B_ROWS),
+                    self._unpack_matrix(self.bus.B, A_COLUMNS_B_ROWS, B_COLUMNS),
+                )
+
+
+@cocotb.coroutine
+def run_test_array(dut, a_data, b_data):
+    """For simulators that don't support multi-dimensional arrays.
+    Matrix M x N:
+     0  [ ]  .  [ ]
+    ...  .   .   .
+    M-1 [ ]  .  [ ]
+         0  ... N-1
+
+    becomes Array 0 .. MxN:
+     [ ] . [ ] .   [ ]
+      0 ... M ... MxN-1
+    """
+    cocotb.fork(Clock(dut.clk, 5, units='ns').start())
+
+    expected_output = []
+
+    def calc_c(transaction):
+        a_array, b_array, c_array = transaction
+
+        c_expected = [
+            BinaryValue(
+                sum(
+                    [
+                        a_array[(i*A_COLUMNS_B_ROWS)+n] * b_array[(n*B_COLUMNS)+j]
```

Comment:
This should be
```suggestion
                        a_array[I][n] * b_array[n][j]
```
Due to the use of unpack_matrix